pkgs/stdenv/adapters.nix

```diff
@@ -171,12 +168,22 @@ rec {
   /* Modify a stdenv so that all buildInputs are implicitly propagated to
      consuming derivations
   */
-  propagateBuildInputs = stdenv:
-    stdenv.override (old: {
-      mkDerivationFromStdenv = extendMkDerivationArgs old (args: {
+  propagateBuildInputs = extendMkDerivationArgs (args: {
         propagatedBuildInputs = (args.propagatedBuildInputs or []) ++ (args.buildInputs or []);
         buildInputs = [];
-      });
+  });
+
+  /* Modify a stdenv so as to extend `mkDerivation`'s arguments.
+     A stronger version of `addAttrsToDerivation`.
+
+     Example:
+      requestCcache = extendMkDerivationArgs
+        (oldAttrs: { requiredSysteFeatures = oldAttrs.requiredSysteFeatures or [ ] ++ [ "ccache" ]; });
+  */
+  extendMkDerivationArgs =
```

Comment:
Fwiw `overrideMkDerivationArgs` would have been a more honest name

Reply:
Hmmm, I think you're right. Internally since `extension` is passed to `overrideAttrs`, it's a more descriptive name which hopefully indicates `extension` can be of the form `final: prev: blarg`.